allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -11,53 +11,57 @@
 
 from allennlp.common import Params
 from allennlp.data.vocabulary import Vocabulary
-from allennlp.data.dataset_readers.encoder_decoder import START_SYMBOL, END_SYMBOL
+from allennlp.data.dataset_readers.seq2seq import START_SYMBOL, END_SYMBOL
 from allennlp.modules import Attention, TextFieldEmbedder, Seq2SeqEncoder
 from allennlp.modules.similarity_functions import SimilarityFunction
 from allennlp.modules.token_embedders import Embedding
 from allennlp.models.model import Model
 from allennlp.nn.util import get_text_field_mask, sequence_cross_entropy_with_logits, weighted_sum
 
 
-@Model.register("encoder_decoder")
-class EncoderDecoder(Model):
+@Model.register("wikitables_parser")
+class WikiTablesSemanticParser(Model):
     """
-    An ``EncoderDecoder`` is a :class:`Model` which takes a sequence, encodes it, and then uses the encoded
-    representations to decode another sequence. It takes an encoder (:class:`Seq2SeqEncoder`) as an
-    input. This class implements the functionality of the decoder.
+    A ``WikiTablesSemanticParser`` is a :class:`Model` which takes as input a table and a question,
+    and produces a logical form that answers the question when executed over the table.  The
+    logical form is generated by a `type-constrained`, `transition-based` parser.  This is a
+    re-implementation of the model used for the paper `Neural Semantic Parsing with Type
+    Constraints for Semi-Structured Tables
+    <https://www.semanticscholar.org/paper/Neural-Semantic-Parsing-with-Type-Constraints-for-Krishnamurthy-Dasigi/8c6f58ed0ebf379858c0bbe02c53ee51b3eb398a>`_,
+    by Jayant Krishnamurthy, Pradeep Dasigi, and Matt Gardner (EMNLP 2017).
 
-    In this implementation, the decoder uses the encoder's outputs in two ways. The hidden state of the decoder
-    is intialized with the output from the final time-step of the encoder, and when using attention, a weighted
-    average of the outputs from the encoder is concatenated to the inputs of the decoder at every timestep.
+    WORK STILL IN PROGRESS.  This is just copying the SimpleSeq2Seq model for now, and we'll
```

Comment:
Since this is work in progress, do we want this in the master branch?

Reply:
Yeah, good point, I'll split it out into a separate branch.